src/neuron/container/view_utils.hpp

```diff
@@ -0,0 +1,138 @@
+ */
+template <typename Identifier>
+struct handle_base {
+    /**
+     * @brief Construct a handle from an identifier.
+     */
+    handle_base(Identifier identifier)
+        : m_identifier{std::move(identifier)} {}
+
+    /**
+     * @brief Return current offset in the underlying storage where this object lives.
+     */
+    [[nodiscard]] std::size_t current_row() const {
+        return m_identifier.current_row();
+    }
+
+    /**
+     * @brief Obtain a lightweight identifier of the current entry.
+     *
+     * The return type is essentially std::size_t* -- it does not contain a
+     * pointer/reference to the actual storage container.
+     */
+    [[nodiscard]] non_owning_identifier_without_container id() const {
+        return m_identifier;
+    }
+
+    /**
+     * @brief This is a workaround for id sometimes being a macro.
+     * @todo Remove those macros once and for all.
+     */
+    [[nodiscard]] auto id_hack() const {
+        return id();
+    }
+
+    /**
+     * @brief Obtain a reference to the storage this handle refers to.
+     */
+    auto& underlying_storage() {
+        return m_identifier.underlying_storage();
+    }
+
+    /**
+     * @brief Obtain a const reference to the storage this handle refers to.
+     */
+    auto const& underlying_storage() const {
+        return m_identifier.underlying_storage();
+    }
+
+  protected:
+    /**
+     * @brief Get a data_handle<T> referring to the given field inside this handle.
+     * @tparam Tag Tag type of the field we want a data_handle to.
+     *
+     * This is used to implement methods like area_handle() and v_handle() in
+     * the interface templates.
+     *
+     * @todo const cleanup -- should there be a const version returning
+     *       data_handle<T const>?
+     */
+    template <typename Tag>
+    [[nodiscard]] auto get_handle() {
+        return underlying_storage().template get_handle<Tag>(this->id());
+    }
+
+    /**
+     * @brief Get a data_handle<T> referring to the (runtime) field_index-th
+     *        copy of a given (static) field.
+     * @tparam Tag Tag type of the set of fields the from which the
+     *             field_index-th one is being requested.
+
+     * @todo Const cleanup as above for the zero-argument version.
+     */
+    template <typename Tag>
+    [[nodiscard]] auto get_handle(int field_index, int array_offset = 0) {
+        return underlying_storage().template get_field_instance_handle<Tag>(this->id(),
+                                                                            field_index,
+                                                                            array_offset);
+    }
+    template <typename Tag>
+    [[nodiscard]] auto& get() {
+        return underlying_storage().template get<Tag>(current_row());
+    }
+    template <typename Tag>
+    [[nodiscard]] auto const& get() const {
+        return underlying_storage().template get<Tag>(current_row());
+    }
+    template <typename Tag>
```

Comment:
Addressed in 1d0f874439f0fdfc593cbbb020a4a038f979f04c, I added the more verbose documentation to the actual get_tag method that is called, rather than to this thin wrapper. But I added a cross-reference.